### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by kmnip** [isovic/racon#115]
Hi,

I am using version 1.3.2 to polish 2 sequences (27903 bp and 4492 bp long) with 835 nanopore reads, but I encountered a segfault after consensus has completed according to the log.
```
[racon::Polisher::initialize] loaded target sequences
[racon::Polisher::initialize] loaded sequences
[racon::Polisher::initialize] loaded overlaps
[racon::Polisher::initialize] aligned overlap 833/833
[racon::Polisher::initialize] transformed data into windows
[racon::Polisher::polish] generated consensus for window 65/65
Segmentation fault (core dumped)
```
Here is my command for running Racon:
```
racon reads.fa reads2targets.paf.gz targets.fa -u > polished_targets.fa
```
The overlaps were generated with minimap2.

Can you please let me know how I can help you fix this?

Thanks,
Ka Ming

**Comment by kmnip** [comment]
I just found out that the segfault has to do with really long CIGAR string in the PAF file. A new PAF file without CIGAR strings has solved the issue for me!

**Comment by rvaser** [comment]
Hi Ka Ming,
thanks for reporting this. How long was the CIGAR string? The parser expects the whole line to be shorter than 3000 characters.

Best regards,
Robert

**Comment by kmnip** [comment]
Hi Robert,

The problematic line has 7522 characters and the CIGAR string has 7315 characters after `cg:Z:`.

Ka Ming

**Comment by cgjosephlee** [comment]
Hi @kmnip ,

I would like to remind that minimap2 use a less precise mapping approach when not outputting  cigar string (`-c`) in paf format, and may affect racon behavior.
https://github.com/lh3/minimap2/blob/master/FAQ.md#1-alignment-different-with-option--a-or--c

Joseph

**Comment by kmnip** [comment]
Yes, I do see noticable difference in my assembly assessment results.
Thanks for the reminder.
